lib/patch-series.ts

```diff
@@ -178,42 +180,96 @@ export class PatchSeries {
             basedOn,
             cc,
             coverLetter,
-        } = PatchSeries.parsePullRequestDescription(pullRequestDescription);
+        } = await PatchSeries.parsePullRequest(workDir,
+                                               pullRequestTitle,
+                                               pullRequestBody,
+                                               senderName, senderEmail);
 
         if (basedOn && !revParse(basedOn, workDir)) {
             throw new Error(`Cannot find base branch ${basedOn}`);
         }
 
         const publishToRemote = undefined;
 
-        const project = await ProjectOptions.get(workDir, headCommit, cc || [],
+        const project = await ProjectOptions.get(workDir, headCommit, cc,
                                                  basedOn, publishToRemote,
                                                  baseCommit);
 
-        const wrapCoverLetterAtColumn = 76;
         return new PatchSeries(notes, options, project, metadata,
                                rangeDiff,
-                               md2text(coverLetter, wrapCoverLetterAtColumn),
+                               coverLetter,
                                senderName);
     }
 
-    protected static parsePullRequestDescription(description: string): {
+    protected static async parsePullRequest(workDir: string,
+                                            prTitle: string,
+                                            prBody: string,
+                                            senderName?: string,
+                                            senderEmail?: string):
+    Promise <{
+        coverLetter: string,
+        basedOn?: string,
+        cc: string[],
+    }> {
+        // Replace \r\n with \n to simplify remaining parsing.
+        // Note that md2text() in the end will do the replacement anyway.
+        prBody = prBody.replace(/\r\n/g, "\n");
+
+        // Remove template from description (if template exists)
+        try {
+            let prTemplate =
+                await git(["show",
+                           "upstream/master:.github/PULL_REQUEST_TEMPLATE.md"],
+                          { workDir });
+            // Depending on core.autocrlf setting, template may contain \r\n
+            prTemplate = prTemplate.replace(/\r\n/g, "\n");
```

Comment:
Should probably still have the `\r?` on the regex.

Reply:
Not to my understanding.

Previous regex tried to replace `\n` to `\r\n` but keep `\r\n` as is without changing to `\r\r\n`. `\r?\n` served this purpose.

Now, I want to change `\r\n` to `\n`. There's no need to change `\n` to `\n` because that's a no-op.

Reply:
Sorry, I was wrong here.